### DISCUSSION TITLE
Runway support

### DISCUSSION THREAD

**Discussion by lukevmk** [aerni/statamic-advanced-seo#155] [Ideas]
Currently the `SitemapIndex` supports Collections and Taxonomies.

For a project we use Runway (https://runway.duncanmcclean.com) (https://statamic.com/addons/rad-pack/runway) a first party Statamic package. That makes it possible to manage Eloquent models in the Statamic interface and more. 

Are you open for a pull request to support Runway?

**Comment by aerni** [comment]
I don't think I understand your request. I'm not sure what Runway has to do with the `SitemapIndex` supporting Collections and Taxonomies?

**Comment by lukevmk** [comment]
The SitemapIndex supporting the Collections and Taxonomies was just an example and proof that my idea could work.

Our use case is a website for a travel agency. Everything is packed in Statamic as you would expect. But the website also has a ton of accommodations that it has to manage and show (over 4.000). 
So we decided to place them in the database, since Statamic performs really badly with large collections. 

That is where Runway came into play. Runway made it possible to manage the database model of the accommodations in the Statamic interface like a normal entry. 

The last hurdle we have to take is SEO. We use your SEO package for all of our websites, but it does not work with Runway (yet). 
I suggested to make the Runway models work like you made it for the Collections and Taxonomies.

**Comment by aerni** [comment]
I understand. I'm not familiar with Runway. Are the sitemaps the only Advanced SEO feature that doesn't work with Runway? Everything else in the SEO tab works as expected?

I'm generally open to a PR. But given how tightly integrated the sitemaps are with Statamic's collections and taxonomies, I'm not sure of the implications of adding support for Runway models. Maybe a different sitemap driver for Runway would make more sense. Again, I don't know. If you can whip up a PR to show the general direction, that would be helpful.

**Comment by lukevmk** [comment]
Noo, the other stuff does not work either, but I got a work around for that. 

The sitemap was to much work to create myself. That is why I am trying this way :) 

Okay! I will send in a PR with a general direction and let's see from there.
I think Runway support could be a really nice add-on for you package!

**Comment by aerni** [comment]
Ok, I hear you. I'd rather add full support for Runway than just partial support for sitemaps. But again, I have no idea about the implications of this. 

You said that you found a workaround for the other stuff that doesn't work. Have you tried using [custom sitemaps](https://advanced-seo.michaelaerni.ch/usage/sitemaps#custom-sitemaps) as a workaround as well?

**Reply by lukevmk** [reply]
I created a workaround for the SEO field, by just adding the fields to the eloquent model. A bit hacky, but enough for now. 

For me it is to much work (for now) to create full Runway support as you suggest. I also understand that that is what you would prefer. 
It would help us very much if we could just do the sitemap functionality first. 

The custom sitemaps could work as a workaround :)

To give more context: I work for nobears.nl and we roughly already have 30 statamic sites live. And more to come..

**Comment by aerni** [comment]
I explored adding support for Runway, but presently, it’s not feasible to implement without resorting to hacky workarounds. Advanced SEO depends heavily on the `EntryBlueprintFound` event to programmatically inject its fields into the blueprint — this is essential for making the fields appear in the Control Panel and for the augmentation process to function correctly. Unfortunately, Runway doesn’t provide an equivalent event, which means there’s currently no clean way to integrate with the blueprint lifecycle. I still hope to add support in the future.
